pensieve/analysis.py

```diff
@@ -80,49 +88,96 @@ def _get_timelimits_if_ready(
 
         return current_time_limits
 
-    def run(self, experiment: experimenter.Experiment, current_date: datetime, dry_run: bool):
+    def _normalize_name(self, name: str) -> str:
+        return re.sub(r"[^a-zA-Z0-9_]", "_", name)
```

Comment:
It is probably also worth checking whether the name is >1024 characters (this is the allowed limit in BigQuery). And shortening the name if necessary.

Reply:
I think erroring out might be better -- we should never see slugs that long because of telemetry constraints, and if we do, it won't be clear which part of the slug matters, so we should probably just give up.